docs/articles/baseplatform/reproducing-yocto-builds.rst

```diff
@@ -0,0 +1,91 @@
+Reproducing yocto builds
+========================
+
+This document describes how to:
+
+* Configure a ``Yocto`` build so that it can be reproduced.
+* Reproduce such a ``Yocto`` build. The reproduction requires that additional artifacts are archived for the Yocto build that is to be reproduced.
+
+Intro to Yocto build reproducibility
+------------------------------------
+
+A bitbake recipe specifies where the source code of a package should be downloaded from. Because there are several ways to do this, a recipe might not specify a specific version but rather the head of some branch or a "latest"-release. Even though this is not recommended, there is no guarantee that developers will not write recipes this way. This of course affects reproducibility as the maintainer of some upstream project may update the head of the branch or the "latest"-release between builds so that the same version of a recipe yields different results. In addition, in cases when reproducibility of a Yocto build is required, the actual availability of the upstream source code packages might cause a problem, e.g. if a tar-archived is moved to some other URL.
+
+The general solution to guarantee reproducibility is to let the Yocto build environment create tar-archives of the downloaded sources and place these in a local mirror during a build. This local mirror can then be used when reproducing the build, a solution that works in most cases. The exception is when a package's source code is stored in a git repository and the recipe for the package does not point to a specific commit but a branch or tag.
```

Comment:
It's not clear to me why this wouldn't work: if Yocto makes a tar archive of the git repo, than it cannot change, can it?